packages/ember-application/lib/system/application.js

```diff
@@ -693,9 +693,15 @@ var Application = Namespace.extend(DeferredMixin, {
     var namespace = this;
     var initializer;
 
+    var makeGraphValue = function(initializer) {
+      return function() {
+        return initializer.initialize.apply(initializer, arguments);
```

Comment:
Using `apply` is slower than just calling the function and if we refactor the arguments then this may need to be updated anyways (along with many other things).  tldr; I would prefer to just invoke the function with the params that we know.